### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by duytantran**
This PR handles the frontend part of #1.

Would like feedback on everything such as best code practice, semantic commit messages etc.
Possible bug in chrome incognito: after one logs in Microsoft, the page displays "Failed to retrieve events..." before it reloads to events. This bug is fixed through: 
`Settings --> Privacy and security --> Site settings --> Cookies and site data --> set Block third-party cookies to off.`
Possible solution is maybe to include `credentials: 'include'` inside `fetch` options, but there may be a potential security risk.

Contact me for the IDs used in `auth.ts`

### CODE REVIEW

File: src/utils/request.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import Authentication from "../config/auth";
+
+type FetchParams = Parameters<typeof fetch>;
+
+export async function authFetch(url: FetchParams[0], options: FetchParams[1]): Promise<Response>{
+    const tokenResponse = await Authentication.getToken();
+    return await fetch(url, {
+        ...options,
+        headers: {
+            'Content-Type': 'application/json',
```
Comment (anhtin) [inline src/utils/request.ts:10]: This forces all requests to be in JSON, not that we're planning on using another serialization format but just wanted to point that out 😆.

On another note, you generally don't want to override _all_ the headers in the `options` parameter. The caller should be able to add additional headers, with maybe only the `Authorization` header being overriden by this function.

**Review by anhtin — COMMENTED** [review]
Looks good to me 😆 Just need to fill in the missing information and fix the formatting - do you not have the Prettier plugin installed on VSCode?

File: src/index.tsx [hunk 2]
```diff
@@ -3,11 +3,22 @@ import ReactDOM from 'react-dom';
 
 import App from './App';
 import * as serviceWorker from './serviceWorker';
+import Authentication from "./config/auth";
 
-ReactDOM.render(
-  <React.StrictMode>
+
+
+const MadLearning = ():JSX.Element =>{
```
[the remark this answers is not in the document]
Comment (duytantran) [reply src/index.tsx:10]: I do have Prettier and it seems like it is enabled globally as well, but apparently I had to add some lines in settings.json to autoformat after each save.

File: src/utils/request.ts (hunk 1 above)
Comment (duytantran) [reply src/utils/request.ts:10]: Good to know and I agree. It is best practice to let other developers override 'Content-Type' and other headers.

**Review by LERoise — APPROVED** [review]

**Comment by duytantran** [comment]
Need to wait for the changes of the environment variables to be merged before mering this PR. Currently, `src/config/auth.ts` have placeholder values for the IDs

**Comment by martinothamar** [comment]
closing this because of #38 since it builds on this but has lots of changes 🙏

**Pull request closed without merging**